### PR TITLE
backup, restore: Write correct api-version in rawkv backup/restore

### LINKED ISSUE #31121: Check API V2 in BR
This is part of https://github.com/tikv/tikv/issues/10938.

- [ ] BR backup should know the current API version of TiKV and write it in backupmeta.
- [ ] BR restore should check the API version in backupmeta and set it in ingest request to TiKV.

### PR DESCRIPTION
**Pull request by peng1999**
Signed-off-by: Peng Guanwen <pg999w@outlook.com>

### What problem does this PR solve?

Issue Number: close #31121

Problem Summary:

BR backup should know the current API version of TiKV and write it in backupmeta.

### What is changed and how it works?

Only RawKV backup/restore is changed because TiDB data is compatible across API V1 and V2.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by kennytm** [comment]
```
[2021-12-29T08:09:55.616Z] /go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-rc.0.20211223062159-300275dee63e/internal/mockstore/mocktikv/pd.go:70:9: cannot use &pdClient{...} (type *pdClient) as type pd.Client in return argument:

[2021-12-29T08:09:55.616Z] 	*pdClient does not implement pd.Client (missing LoadGlobalConfig method)

[2021-12-29T08:09:55.616Z] # github.com/tikv/client-go/v2/internal/mockstore/mocktikv

[2021-12-29T08:09:55.616Z] /go/pkg/mod/github.com/tikv/client-go/v2@v2.0.0-rc.0.20211223062159-300275dee63e/internal/mockstore/mocktikv/pd.go:70:9: cannot use &pdClient{...} (type *pdClient) as type pd.Client in return argument:

[2021-12-29T08:09:55.616Z] 	*pdClient does not implement pd.Client (missing LoadGlobalConfig method)

```

File: go.mod [hunk 1]
```diff
@@ -103,3 +103,6 @@ replace github.com/pingcap/tidb/parser => ./parser
 
 // fix potential security issue(CVE-2020-26160) introduced by indirect dependency.
 replace github.com/dgrijalva/jwt-go => github.com/form3tech-oss/jwt-go v3.2.6-0.20210809144907-32ab6a8243d7+incompatible
+
+// See https://github.com/pingcap/tidb/pull/31010
+replace github.com/tikv/client-go/v2 => github.com/lemonhx/client-go/v2 v2.0.0-rc.0.20211224062518-60bb3ea439ab
```
Comment (kennytm) [inline go.mod:108]: does this mean this PR is blocked by #31010?

**Review by kennytm — COMMENTED** [review]
rest lgtm

File: go.mod (hunk 1 above)
Comment (peng1999) [reply go.mod:108]: This is blocked by https://github.com/tikv/client-go/pull/405. The root cause of the compilation error is https://github.com/pingcap/kvproto/pull/828.

**Comment by peng1999** [comment]
If #31219 (likely to get merged early than https://github.com/tikv/client-go/pull/405) gets merged, this one will also be unblocked.

**Review by andylokandy — APPROVED** [review]

**Comment by peng1999** [comment]
@kennytm The blockers have merged, could you please look at the PR again?

**Review by kennytm — APPROVED** [review]
please resolve conflict

**Review by andylokandy — APPROVED** [review]

**Comment by Mini256** [comment]
/unhold

It seems ready to merge.

**Pull request merged**